goss-testing/scripts/postgres_replication_lag.sh

```diff
@@ -25,19 +62,54 @@ do
     first_member="$(kubectl get pod -n $c_ns -l "cluster-name=$c_name,application=spilo" \
                   -o custom-columns=NAME:.metadata.name --no-headers | head -1)"
 
-    for lag in $(kubectl -n $c_ns exec $first_member -- patronictl list --format json 2>/dev/null | jq '.[]."Lag in MB"')
-    do
-        if [[ $lag != "\"\"" ]] && [[ $lag == "\"unknown"\" || $lag -gt 50 ]]; 
-        then 
-            if [[ $print_results -eq 1 ]]; 
-            then 
-                echo "Warning: $c_name has Lag: $lag"; 
-                failFlag=1 
+    echo -n "$c_name - "
+    c_attempt=0
+    c_lag_history=""
+    while [ true ]; do
+        c_attempt=$((${c_attempt} + 1))
+
+        if [[ $c_attempt -gt 1 ]]; then
+            # Sleep before re-attempting
+            sleep $POSTGRES_WAIT_SECONDS_BETWEEN_ATTEMPTS
+        fi
+
+        cluster_lag=$(kubectl exec $first_member -c postgres -it -n ${c_ns} -- curl -s http://localhost:8008/cluster | jq '[.members[] | .lag]')
+        c_max_lag=$(echo $cluster_lag | jq max)
+        c_unknown_lag=$(echo $cluster_lag | grep "unknown" | wc -l)
+        if [[ -n $c_lag_history ]]; then
+            c_lag_history="${c_lag_history}, $c_max_lag"
+        else
+            c_lag_history="$c_max_lag"
+        fi
+
+        #check lag:unknown
+        if [[ $c_unknown_lag -gt 0 ]]; then
+            echo -e "\n--- ERROR --- $c_name cluster has lag: unknown"
+            if [[ $print_results -eq 1 ]]
+            then failFlag=1
                 kubectl -n $c_ns exec $first_member -- patronictl list 2>/dev/null
+                break;
             else exit 1; fi
         fi
+
+        if [[ $POSTGRES_MAX_LAG =~ ^[0-9][0-9]*$ ]]; then
```

Comment:
This script would have exited at line 49 if `! $POSTGRES_MAX_LAG =~ ^[0-9][0-9]*$` so this check seems redundant.

Reply:
Removed check